api/src/messages/services/messages.service.ts

```diff
@@ -45,11 +36,19 @@ export class MessagesService {
         if (!result) throw new NotFoundException();
     }
 
-    async update(id: number, updateMessageDto: UpdateMessageDto) {
-        const message = await this.findOne(id);
+    async update(
+        id: number,
+        updateMessageDto: UpdateMessageDto,
+        isImage: boolean
+    ) {
+        const messageIsImage = updateMessageDto.isImage;
```

Comment:
This method and controller are very convoluted. Simplify this.
1. Return early pattern. Check at first `updateMessageDto.isImage` or `!message.isImage && updateMessageDto.isImage` and return
2. `        if (!this.isDevelopment && message.author.id !== userId)
            throw new ForbiddenException();` at the begninig of funtion not after db quary
3. Remove meaningless variables.
4. Better naming for variables because this code is very hard to read.

Reply:
1. U don't understand what i mean here, i can explain more on DC
2. Watch the code, i can't before, because i need that query to run this "if" :)
3. Which var?
4. Also which?

Reply:
Done